pkg/operator/staticpod/controller/monitoring/monitoring_resource_controller.go

```diff
@@ -0,0 +1,204 @@
+	case operatorv1.Unmanaged:
+		return nil
+	case operatorv1.Removed:
+		// TODO: Should we try to actively remove the resources created by this controller here?
+		return nil
+	}
+
+	directResourceResults := resourceapply.ApplyDirectly(c.kubeClient, c.eventRecorder, c.mustTemplateAsset,
+		"manifests/prometheus-role.yaml",
+		"manifests/prometheus-role-binding.yaml",
+	)
+
+	errs := []error{}
+	for _, currResult := range directResourceResults {
+		if currResult.Error != nil {
+			errs = append(errs, fmt.Errorf("%q (%T): %v", currResult.File, currResult.Type, currResult.Error))
+		}
+	}
+
+	serviceMonitorBytes, err := c.mustTemplateAsset("manifests/service-monitor.yaml")
+	if err != nil {
+		errs = append(errs, fmt.Errorf("manifests/service-monitor.yaml: %v", err))
+	} else {
+		_, serviceMonitorErr := resourceapply.ApplyServiceMonitor(c.dynamicClient, c.eventRecorder, serviceMonitorBytes)
+		errs = append(errs, serviceMonitorErr)
+	}
+
+	err = common.NewMultiLineAggregate(errs)
+
+	// NOTE: Failing to create the monitoring resources should not lead to operator failed state.
+	cond := operatorv1.OperatorCondition{
+		Type:   operatorStatusMonitoringResourceControllerFailing,
+		Status: operatorv1.ConditionFalse,
+	}
+	if err != nil {
+		cond.Status = operatorv1.ConditionTrue
+		cond.Reason = "Error"
+		cond.Message = err.Error()
+	}
+	if _, updateError := common.UpdateStatus(c.operatorConfigClient, common.UpdateConditionFn(cond)); updateError != nil {
+		if err == nil {
+			return updateError
+		}
+	}
+
+	return err
+}
+
+func (c *MonitoringResourceController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	glog.Infof("Starting MonitoringResourceController")
+	defer glog.Infof("Shutting down MonitoringResourceController")
+	if !cache.WaitForCacheSync(stopCh, c.saListerSynced) {
+		return
+	}
+	if !cache.WaitForCacheSync(stopCh, c.clusterRoleBindingListerSynced) {
+		return
+	}
+
+	// doesn't matter what workers say, only start one.
+	go wait.Until(c.runWorker, time.Second, stopCh)
+
+	<-stopCh
+}
+
+func (c *MonitoringResourceController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *MonitoringResourceController) processNextWorkItem() bool {
+	dsKey, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(dsKey)
+
+	err := c.sync()
+	if err == nil {
+		c.queue.Forget(dsKey)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", dsKey, err))
+	c.queue.AddRateLimited(dsKey)
+
+	return true
+}
+
+// eventHandler queues the operator to check spec and status
+func (c *MonitoringResourceController) eventHandler() cache.ResourceEventHandler {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.queue.Add(controllerWorkQueueKey) },
+		UpdateFunc: func(old, new interface{}) { c.queue.Add(controllerWorkQueueKey) },
+		DeleteFunc: func(obj interface{}) { c.queue.Add(controllerWorkQueueKey) },
+	}
+}
```

Comment:
anything in here that is not copied from the other controllers?

Reply:
this is basically a backing resource controller with some extra logic around service monitor (resourceapply.ApplyDirect won't work)... Also failure to create monitoring resources should not set the operator to failed state.